schematic/store/synapse.py

```diff
@@ -538,6 +538,94 @@ def getDataTypeFromManifest(self, manifestId:str):
 
         return result_dict
 
+    def _get_files_metadata_from_dataset(self, datasetId: str, only_new_files: bool, manifest:pd.DataFrame=None) -> Optional[dict]:
+        """retrieve file ids under a particular datasetId
+
+        Args:
+            datasetId (str): a dataset id 
+            only_new_files (bool): if only adding new files that are not already exist 
+            manifest (pd.DataFrame): metadata manifest dataframe. Default to None. 
+
+        Returns:
+            a dictionary that contains filename and entityid under a given datasetId or None if there is nothing under a given dataset id are not available
+        """
+        dataset_files = self.getFilesInStorageDataset(datasetId)
+        if dataset_files:
+            dataset_file_names_id_dict = self._get_file_entityIds(dataset_files, only_new_files=only_new_files, manifest=manifest)
+            return dataset_file_names_id_dict
+        else:
+            return None
+
+    def add_entity_id_and_filename(self,  datasetId: str, manifest: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Not that we have to implement now, but do you think in the future it would be useful to add one method as a wrapper that calls either `add_entity_id_and_filename` or `fill_in_entity_id_filename` depening on the content of the manifest so that users (other devs) don't have to worry about if the columns exist in the manifest before hand?

Reply:
Yes definitely. I am trying not to make this PR changes too many things, so I don't want to refactor this part. But I completely agree that in the future, how we fill up `filename` and `entityId` column should be the same across code base.

Reply:
See https://sagebionetworks.jira.com/browse/FDS-958

Reply:
@linglp keeping the changed in this PR more focused makes sense; thanks for creating the issue!